### PR TITLE
added ignore-case option

### PR DESCRIPTION
**Pull request by donvini94**
Added the option to provide the -i flag to wofi-pass to enable case insensitive search.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Usage: wofi-pass [options]
 	-c, --copy [cmd]	copy to clipboard. Defaults to wl-copy if no cmd is given.
 	-f, --fileisuser	use the name of the password file as username
 	-h, --help		show this help message
+	-i, --ignore-case		enable case insensitive search
```
Comment (schmidtandreas) [inline README.md]: I would prefer to use the same long argument as `wofi` is using. Please rename it in `--insensitive`

File: wofi-pass [hunk 2]
```diff
@@ -217,7 +219,11 @@ function get_passname_from_menu() {
     local wofi_ret
     local wofi_args=()
 
-    wofi_args+=("--dmenu")
+    # Set up Wofi arguments for dmenu mode and optional case-insensitive search
+    wofi_args+=("--show" "dmenu")
+    if [ ${FLAG_CASE_INSENSITIVE} -eq 1 ]; then
+        wofi_args+=("-i") # Enable case-insensitive search if the flag is set
```
Comment (schmidtandreas) [inline wofi-pass]: Please use the long argument, and then you can remove the comment behind the line. I think it is clear then what is does.

File: wofi-pass [hunk 3]
```diff
@@ -217,7 +219,11 @@ function get_passname_from_menu() {
     local wofi_ret
     local wofi_args=()
 
-    wofi_args+=("--dmenu")
+    # Set up Wofi arguments for dmenu mode and optional case-insensitive search
+    wofi_args+=("--show" "dmenu")
```
Comment (schmidtandreas) [inline wofi-pass]: Why do you change the argument from `--dmenu` to `--show dmenu` what is the benefit of it?

File: wofi-pass [hunk 4]
```diff
@@ -169,7 +170,7 @@ load_config_file () {
 }
 
 function parse_arguments() {
```
Comment (schmidtandreas) [inline wofi-pass:172]: Please extends the test for parse_arguments and check if `--insensivite` is set the FLAG_CASE_INSENSITIVE. See tests/test_args.

File: wofi-pass [hunk 5]
```diff
@@ -217,7 +219,11 @@ function get_passname_from_menu() {
     local wofi_ret
     local wofi_args=()
```
Comment (schmidtandreas) [inline wofi-pass:221]: Please extends the test for get_passname_from_menu and check if `--insensitive` really passed to wofi. See tests/test_get_passname

**Review by schmidtandreas — COMMENTED** [review]
I like the idea. Please add tests, so that I can release the `wofi-pass` in reliable way.

File: wofi-pass (hunk 3 above)
Comment (donvini94) [reply wofi-pass]: Without it, I couldn't pass the insensitive to wofi and received an error. With it, in my tests everything existing worked + I could pass the insentivity

**Comment by donvini94** [comment]
Updated as suggested, not entirely sure about the tests in `test_get_passname`, as it is my first time testing bash scripts.

File: wofi-pass (hunk 3 above)
Comment (schmidtandreas) [reply wofi-pass]: I looked to the implementation of wofi. There are no difference between `--dmenu` or `--show dmenu`. So what exactly is not working? Could you provide some logs? What is the wofi version, that are you using?

I tested it with `--dmenu` and works on to different environments.

Without reasons, I am not agree with this change.

File: README.md (hunk 1 above)
Comment (schmidtandreas) [reply README.md]: Please adjust the readme

Comment (schmidtandreas) [reply README.md]: I missing the description of the new parameter in the man page. Please add it to `man/wofi-pass.1.ronn`

File: wofi-pass (hunk 3 above)
Comment (donvini94) [reply wofi-pass]: You were right, there is no difference, I just checked it again. Apparently I only used `dmenu` without `--` for some reason. It works without the change and I reverted it.

File: wofi-pass (hunk 5 above)
Comment (donvini94) [reply wofi-pass:221]: all tests successful on my end

File: README.md [hunk 6]
```diff
@@ -6,7 +6,8 @@ Usage: wofi-pass [options]
 	-a, --autotype		autotype whatever entry is chosen
 	-c, --copy [cmd]	copy to clipboard. Defaults to wl-copy if no cmd is given.
 	-f, --fileisuser	use the name of the password file as username
-	-h, --help		show this help message
+	-h, --help			show this help message
```
Comment (schmidtandreas) [inline README.md]: why do you changed this line?

File: README.md [hunk 7]
```diff
@@ -6,7 +6,8 @@ Usage: wofi-pass [options]
...
+	-i, --ignore-case	enable case insensitive search
```
Comment (schmidtandreas) [inline README.md]: Still wrong

File: README.md [hunk 8]
```diff
@@ -6,7 +6,8 @@ Usage: wofi-pass [options]
...
 	-s, --squash		don't show field choice if password file only contains password
 	-t, --type [cmd]	type the selection instead of copying to clipboard.
 				Defaults to wtype if no cmd is given.
```
Comment (schmidtandreas) [inline README.md:13]: Please add man page description for new parameter in `man/wofi-pass.1.ronn`

File: tests/test_get_passname [hunk 9]
```diff
@@ -25,6 +25,22 @@ test_passname_selection() {
...
+    fake find fake_find "${HOME}/.password-store" "${items[@]}"
+    fake _wofi fake_wofi ${select_item}
+
+    local output="$(get_passname_from_menu "${FAKE_PARAMS[@]}")"
+
+    assert_equals "${items[${select_item}]}" "${output}" "passname: failed to find item in case-insensitive search"
+
+    FAKE_PARAMS=()
+}
+
```
Comment (schmidtandreas) [inline tests/test_get_passname:49]: This makes not much sense.
Here we should just check if `--insensitive` args is passed to `wofi` if FLAG_CASE_INSENSITIVE is 1.

Suggestion:
```
is_arg_set() {
    local regex=".*--${2}.*"
    [[ "${1}" =~ ${regex} ]] && return 0 || return 1
}

test_passname_case_insensetive_args() {
    local result
    fake find echo ""
    fake _wofi wofi_args
    local output="$(get_passname_from_menu)"

    result=true
    is_arg_set "${output}" "insensitive" && result=false
    assert ${result} "passname: wofi argument insensitive was unexpected passed"

    FLAG_CASE_INSENSITIVE=1
    output="$(get_passname_from_menu)"
    result=false
    is_arg_set "${output}" "insensitive" && result=true
    assert ${result} "passname: wofi argument insensitive not found"
}
```

**PLEASE TEST THIS EXAMPLE CODE**

**Review by schmidtandreas — CHANGES_REQUESTED** [review]

File: wofi-pass [hunk 10]
```diff
@@ -217,7 +219,11 @@ function get_passname_from_menu() {
...
 
+    # Set up Wofi arguments for dmenu mode and optional case-insensitive search
```
Comment (schmidtandreas) [inline wofi-pass]: Please revert this comment also. This code does not really need explanations.

File: tests/test_get_passname (hunk 9 above)
Comment (donvini94) [reply tests/test_get_passname:49]: I did test your example code and they pass. The intention of my test was not to test whether insensitive flag is passed, but whether passing the insensitive flag actually results in finding passwords by not searching for them in a case sensitive way. 

As mentioned earlier, I never wrote unit tests for bash scripts. I found this program, thought it's nice but missing a critical feature, and just wanted to add that.

**Review by schmidtandreas — APPROVED** [review]

**Comment by schmidtandreas** [comment]
Thanks for implementation :)

**Pull request merged**